### PR TITLE
Limit  to the bounds of MapView.

### PR DESCRIPTION
**Pull request by pengdev**
### Summary of changes

Android counterpart of https://github.com/mapbox/mapbox-maps-ios/pull/1195

Currently we cannot guarantee that the returned point of `MapboxMap#pixelForCoordinate` is valid outside of the MapView bounds. Originally the underlying implementation was designed to convert points only on the visible parts of the map. Until the GL-Native implementation is reworked we have to limit the public API.

## Pull request checklist:
 - [x] Briefly describe the changes in this PR.
 - [ ] Include before/after visuals or gifs if this PR includes visual changes.
    
 - [x] Write tests for all new functionality. If tests were not written, please explain why.
 - [ ] Optimize code for java consumption (`@JvmOverloads`, `@file:JvmName`, etc).
 - [ ] Add example if relevant.
 - [ ] Document any changes to public APIs.
 - [ ] Run `make update-api` to update generated api files, if there's public API changes, otherwise the `verify-api-*` CI steps might fail.
 - [x] Update [CHANGELOG.md](../CHANGELOG.md) or use the label 'skip changelog', otherwise `check changelog` CI step will fail.
 - [ ] If this PR is a `v10.[version]` release branch fix / enhancement, merge it to `main` firstly and then port to `v10.[version]` release branch.

Fixes: < Link to related issues that will be fixed by this pull request, if they exist >

PRs must be submitted under the terms of our Contributor License Agreement [CLA](https://github.com/mapbox/mapbox-maps-android/blob/main/CONTRIBUTING.md#contributor-license-agreement).

### CODE REVIEW

File: sdk/src/test/java/com/mapbox/maps/MapboxMapTest.kt [hunk 1]
```diff
@@ -583,10 +583,31 @@ class MapboxMapTest {
...
+      nativeMap.pixelForCoordinate(point)
+      nativeMap.size
+    }
+    assertEquals(ScreenCoordinate(-1.0, -1.0), screenCoordinate)
+  }
+
+  @Test
+  fun pixelForCoordinateWithinMapView() {
+    val point = mockk<Point>()
+    val convertedScreenCoordinate = ScreenCoordinate(10.0, 10.0)
```
Comment (alexander-kulikovskii) [inline sdk/src/test/java/com/mapbox/maps/MapboxMapTest.kt:602]: nice to add (0,0) and (width, height) coordinates as a corner cases.
I guess it can be done with parameterized class. use one map size and change only coordinates

**Review by endanke — APPROVED** [review]
Looks good, thanks for implementing it! % comments above

**Review by ank27 — APPROVED** [review]

File: sdk/src/main/java/com/mapbox/maps/MapboxMap.kt [hunk 2]
```diff
@@ -663,12 +663,22 @@ class MapboxMap :
    *
    * Map must be fully loaded for getting an altitude-compliant result if using 3D terrain.
    *
+   * If the screen coordinate is outside of the bounds of `MapView` the returned screen coordinate
+   * contains -1 for both coordinates.
+   *
    * @param coordinate A geographical coordinate on the map to convert to a screen coordinate.
    *
-   * @return Returns a screen coordinate on the screen in [MapOptions.size] platform pixels.
+   * @return Returns a screen coordinate on the screen in [MapOptions.size] platform pixels. If the screen coordinate is outside of the bounds of `MapView` the returned screen coordinate contains -1 for both coordinates.
```
Comment (kiryldz) [inline sdk/src/main/java/com/mapbox/maps/MapboxMap.kt:671]:
```suggestion
   * @return Returns a screen coordinate on the screen in [MapOptions.size] platform pixels. If the screen coordinate is outside of the bounds of [MapView] the returned screen coordinate contains -1 for both coordinates.
```

File: CHANGELOG.md [hunk 3]
```diff
@@ -9,6 +9,7 @@ Mapbox welcomes participation and contributions from everyone.
   Users check the status of MapboxMap and Style by new method MapboxMap.isValid() and Style.isValid() to avoid such exception.
   Automatically unsubscribe all observers in MapboxMap when `MapboxMap.onDestroy()` is invoked. ([1193](https://github.com/mapbox/mapbox-maps-android/pull/1193)) ([1202](https://github.com/mapbox/mapbox-maps-android/pull/1202))
 * Add `MapboxMap#coordinateBoundsForCameraUnwrapped` method for API consistency. ([1222](https://github.com/mapbox/mapbox-maps-android/pull/1222))
+* Limit `MapboxMap#pixelForCoordinate` to the bounds of MapView. ([#1226](https://github.com/mapbox/mapbox-maps-ios/pull/1226))
```
Comment (kiryldz) [inline CHANGELOG.md:12]: link leads to ios 😆

**Review by kiryldz — APPROVED** [review]
requires rebase, dropped some comments but in general LGTM

File: sdk/src/main/java/com/mapbox/maps/MapboxMap.kt [hunk 4]
```diff
@@ -663,12 +663,22 @@ class MapboxMap :
    *
    * Map must be fully loaded for getting an altitude-compliant result if using 3D terrain.
    *
+   * If the screen coordinate is outside of the bounds of `MapView` the returned screen coordinate
```
[the remark this answers is not in the document]
Comment (pengdev) [reply sdk/src/main/java/com/mapbox/maps/MapboxMap.kt:743]: yes

File: sdk/src/test/java/com/mapbox/maps/MapboxMapTest.kt (hunk 1 above)
Comment (pengdev) [reply sdk/src/test/java/com/mapbox/maps/MapboxMapTest.kt]: refactored to parameterized tests
